agent/engine/docker_task_engine_test.go

```diff
@@ -156,6 +156,7 @@ func TestBatchContainerHappyPath(t *testing.T) {
 	default:
 	}
 	eventsReported.Wait()
+	client.EXPECT().DescribeContainer(gomock.Any()).AnyTimes()
```

Comment:
hmmm,   it seems that CheckTaskState() did NOT gets called as part of "steady state verification".

I have added following "debug", I did NOT see log which indicate CheckTaskState() is get called during unit test.

diff --git a/agent/engine/task_manager.go b/agent/engine/task_manager.go
index 3e23355..7be8c97 100644
--- a/agent/engine/task_manager.go
+++ b/agent/engine/task_manager.go
@@ -127,7 +127,7 @@ func (mtask *managedTask) overseeTask() {
        for {
                // If it's steadyState, just spin until we need to do work
                for mtask.steadyState() {
-                       llog.Debug("Task at steady state", "state", mtask.KnownStatus.String())
+                       llog.Info("Task at steady state", "state", mtask.KnownStatus.String())
                        maxWait := make(chan bool, 1)
                        timer := mtask.time().After(steadyStateTaskVerifyInterval)
                        go func() {
@@ -135,9 +135,9 @@ func (mtask *managedTask) overseeTask() {
                                maxWait <- true
                        }()
                        timedOut := mtask.waitEvent(maxWait)
-
+                       llog.Info("liwwu: after timer")
                        if timedOut {
-                               llog.Debug("Checking task to make sure it's still at steadystate")
+                               llog.Info("!!!liwwu: Checking task to make sure it's still at steadystate")
                                go mtask.engine.CheckTaskState(mtask.Task)
                        }
                }